core/internal/server/features.go

```diff
@@ -0,0 +1,89 @@
+package server
```

Comment:
i think the server package name could be confusing 
maybe something more related to the fact that it is feature checker or whatever...

Reply:
Agreed sever was a little confusing. Initially used server since that was internal interactions with server.

featurechecker package seems reasonable as well to me.